### DISCUSSION TITLE
Required Parameters for TMAP8 Nodal Kernels

### DISCUSSION THREAD

**Discussion by acunningham23** [idaholab/TMAP8#46] [Q&A] [answered]
Hi, I’m fairly new to TMAP8 and have a few queries regarding the TMAP-8 only syntax, documented at https://mooseframework.inl.gov/TMAP8/syntax/tmap_only.html 

**ReleasingNodalKernel ($\alpha_{r} C_{t}$):**
Q1) As mentioned in the documentation, the releasing rate coefficient $\alpha_{r}$ may be the following function of temperature: $$\alpha_{r} = \alpha_{r0} \ \mathrm{exp}{(-\epsilon/T)}. $$ Under ‘Required Parameters’ for this kernel, there is a parameter ‘alpha_r’ described as "the release rate coefficient".  However, also under the 'Required Parameters' are ‘temp’ and ‘trapping_energy’, which wouldn't seem to be necessary to define if the user has already defined the "full" $\alpha_{r}$. In this case then, is ‘alpha_r’ referring to the pre-exponential factor $\alpha_{r0}$ rather than $\alpha_{r}$?

Q2) In verification case 1d, the parameter ‘temp’ for this nodal kernel is assigned a constant value of 1000. Is there a way to have the ‘temp’ parameter vary over time?

**TrappingNodalKernel:**
Q3) Under ‘Required Parameters’ for this kernel, there is a parameter ‘alpha_t’ described as "the trapping rate coefficient". Similar to the above kernel, does this refer to $\alpha_{t}$ (which can be defined by a similar exponential function of temperature to $\alpha_{r}$) or the pre-exponential factor $\alpha_{t0}$ (in which case does the user also need to be able to define additional parameters like temperature as with ReleasingNodalKernel)? 

I’d really appreciate any support with the above, and happy to provide more context/information if it would help. Thanks in advance!

**Comment by singhgp4321** [comment, marked as answer]
@acunningham23  Thank you for your question. Sorry for the delayed response.

1) Yes, you are right 'alpha_r' is referring to the pre-exponential factor. I will make a modification to the code to remove this potential confusion.

You can look at https://github.com/idaholab/TMAP8/blob/devel/src/nodal_kernels/ReleasingNodalKernel.C to see how _alpha_r is being used. Note that in line 25 the value assigned by the user to parameter alpha_r is copied to the variable _alpha_r which is then used elsewhere.

2) Yes, you can do it in the following manner, for example. These blocks will need to be added to the input file.

[AuxVariables]
  [temp]
    initial_condition = 300
  []
[]

[Functions]
  [temp_bc_func]
    type = ParsedFunction
    value = '300 + 100*t/3600'  # an example function to vary temperature as a function of time t. You can even define as a function of spatial coordinates x, y, z if needed.
[]

[AuxKernels]
  [temp_aux]
    type = FunctionAux
    variable = temp
    function = temp_bc_func
  []
[]

and inside the [release] block you would simply do "temp = temp" instead of "temp = 1000". The above three blocks are setting up a variable 'temp' whose value is being computed by the temp_aux auxkernel according the function temp_bc_func.

3) Please take a look at https://github.com/idaholab/TMAP8/blob/devel/src/nodal_kernels/TrappingNodalKernel.C to see how _alpha_t is being used. You'll note that the entered value for 'alpha_t' parameter is being used as it is without any exponential term multiplication.

Let me know if you have further question. Again, sorry for the delayed response.

**Reply by acunningham23** [reply]
Apologies for my delayed reply, but thank you very much for your detailed response! All really helpful and clear
